subprojects/core-api/src/main/java/org/gradle/api/DomainObjectCollection.java

```diff
@@ -58,6 +58,8 @@ public interface DomainObjectCollection<T> extends Collection<T> {
      * Returns a collection containing the objects in this collection of the given type.  The returned collection is
      * live, so that when matching objects are later added to this collection, they are also visible in the filtered
      * collection.
+     * <p>
+     * This method is an intermediate operation. It is neither lazy nor eager by itself.
```

Comment:
What does that mean? Can it cause realization under certain conditions?

Reply:
It gives you a collection without semantics change. What you do with that collection can be lazy or eager.

Reply:
Let's just say this then. "This method is an intermediate operation, it doesn't change the realized/unrealized state of the elements in the collection."